analysis/metric_space.lean

```diff
@@ -92,6 +92,19 @@ by rw dist_comm z; apply dist_triangle
 theorem dist_triangle_right (x y z : α) : dist x y ≤ dist x z + dist y z :=
 by rw dist_comm y; apply dist_triangle
 
+lemma dist_triangle4 (x y z t : α) :
+  dist x t ≤ dist x y + dist y z + dist z t :=
+calc
+  dist x t ≤ dist x z + dist z t : dist_triangle x z t
+       ... ≤ (dist x y + dist y z) + dist z t : add_le_add_right (metric_space.dist_triangle x y z) _
+
+lemma dist_triangle5 (x y z t u : α) :
```

Comment:
What's this for? It doesn't get used in this commit. I can see a use for `dist_triangle4`, possibly with the last `dist` swapped, since that type of epsilon/3 argument comes up often, but I would prefer to see these only added as needed because there is an infinite family of such things and they are mostly unhelpful.

Reply:
Agreed. This is a leftover from futher developments, that has nothing to do in this branch (although I know by experience that I use it over and over again). I have removed it, and I will add it back when really needed.